bigquery/storage/managedwriter/adapt/protoconversion.go

```diff
@@ -0,0 +1,259 @@
+						haveDep = true
+						break
+					}
+				}
+				// if dep is missing, add to current dependencies
+				if !haveDep {
+					deps = append(deps, foundDesc.ParentFile())
+				}
+				// construct field descriptor for the message
+				fdp, err := tableFieldSchemaToFieldDescriptorProto(f, fNumber, string(foundDesc.FullName()))
+				if err != nil {
+					return nil, fmt.Errorf("couldn't convert to FieldDescriptorProto (%s): %v", currentScope, err)
+				}
+				fields = append(fields, fdp)
+			} else {
+				// Wrap the current struct's fields in a TableSchema outer message, and then build the submessage.
+				ts := &storagepb.TableSchema{
+					Fields: f.GetFields(),
+				}
+				desc, err := storageSchemaToDescriptorInternal(ts, currentScope, cache)
+				if err != nil {
+					return nil, fmt.Errorf("couldn't compile (%s): %v", currentScope, err)
+				}
+				// Now that we have the submessage definition, we append it both to the local dependencies, as well
+				// as inserting it into the cache for possible reuse elsewhere.
+				deps = append(deps, desc.ParentFile())
+				err = cache.add(ts, desc)
+				if err != nil {
+					return nil, fmt.Errorf("failed to add descriptor to cache: %v", err)
+				}
+				fdp, err := tableFieldSchemaToFieldDescriptorProto(f, fNumber, currentScope)
+				if err != nil {
+					return nil, fmt.Errorf("couldn't compute field schema (%s): %v", currentScope, err)
+				}
+				fields = append(fields, fdp)
+			}
+		} else {
+			fd, err := tableFieldSchemaToFieldDescriptorProto(f, fNumber, currentScope)
+			if err != nil {
+				return nil, err
+			}
+			fields = append(fields, fd)
+		}
+	}
+	// Start constructing a DescriptorProto.
+	dp := &descriptorpb.DescriptorProto{
+		Name:  proto.String(scope),
+		Field: fields,
+	}
+
+	// Use the local dependencies to generate a list of filenames.
+	depNames := []string{
+		wellKnownTypesWrapperName,
+	}
+	for _, d := range deps {
+		depNames = append(depNames, d.ParentFile().Path())
+	}
+
+	// Now, construct a FileDescriptorProto.
+	fdp := &descriptorpb.FileDescriptorProto{
+		MessageType: []*descriptorpb.DescriptorProto{dp},
+		Name:        proto.String(fmt.Sprintf("%s.proto", scope)),
+		Syntax:      proto.String("proto3"),
+		Dependency:  depNames,
+	}
+
+	// We'll need a FileDescriptorSet as we have a FileDescriptorProto for the current
+	// descriptor we're building, but we need to include all the referenced dependencies.
+	fds := &descriptorpb.FileDescriptorSet{
+		File: []*descriptorpb.FileDescriptorProto{
+			fdp,
+			protodesc.ToFileDescriptorProto(wrapperspb.File_google_protobuf_wrappers_proto),
+		},
+	}
+	for _, d := range deps {
+		fds.File = append(fds.File, protodesc.ToFileDescriptorProto(d))
+	}
+
+	// Load the set into a registry, then interrogate it for the descriptor corresponding to the top level message.
+	files, err := protodesc.NewFiles(fds)
+	if err != nil {
+		return nil, fmt.Errorf("protodesc.NewFiles: %v", err)
```

Comment:
This error message does not seem to add too much context, should the original error just be returned?

Reply:
done